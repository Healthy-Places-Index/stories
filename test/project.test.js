import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MockedProvider } from '@apollo/client/testing';
import * as nextRouter from 'next/router';
import Create, { GET_PROJECT } from '../src/pages/project/[project]';

const mocks = [
  {
    request: {
      query: GET_PROJECT,
      variables: {
        project: '1',
      },
    },
    result: {
      data: {
        Project: {
          id: '1',
          title: 'Test Map',
          description: 'Description',
          tags: [{ id: '2' }],
          category: 'Literature',
          imageTitle: 'Test image',
          creator: 'Test creator',
          source: 'Test source',
          url: 'https://rio-narrative-files.s3.amazonaws.com/',
          published: true,
        },
        categories: {
          values: [
            {
              key: 'Literature',
              text: 'Literature',
              value: 'Literature',
            },
          ],
        },
        tags: [
          {
            key: '2',
            text: 'Test',
            value: '2',
          },
        ],
      },
    },
  },
];

nextRouter.useRouter = jest.fn();
nextRouter.useRouter.mockImplementation(() => ({ query: { project: '1' } }));

jest.mock('semantic-ui-react', () => {
  const Button = props => <div {...props} className="button" />;
  const Field = props => <div {...props} className="field" />;
  const Form = { Field };
  const Input = props => <input {...props} className="input" />;
  const Dimmer = props => <div {...props} className="dimmer" />;
  const Loader = props => <div {...props} className="loader" />;
  return { Button, Form, Input, Dimmer, Loader };
});

jest.mock('semantic-ui-react/dist/commonjs/addons/Portal/Portal', () => ({ children }) => children);

describe('project page', () => {
  it('matches snapshot', async () => {
    const component = renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Create />
      </MockedProvider>
    );

    await act(() => new Promise(resolve => setTimeout(resolve, 0)));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
