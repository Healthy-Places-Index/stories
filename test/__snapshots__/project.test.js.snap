// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project page matches snapshot 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#FAFAFA",
      "minHeight": "100vh",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <div
      className="header"
    >
      <img
        className="img"
        src="/img/rio-logo.svg"
        style={
          Object {
            "width": 150,
          }
        }
      />
      <div
        className="spacer"
      />
      <a
        className="link"
        href="undefined/"
      >
        Home
      </a>
      <a
        className="link"
        href="undefined/about"
      >
        About
      </a>
      <a
        className="link"
        href="undefined/people"
      >
        People
      </a>
      <a
        className="link"
        href="undefined/research"
      >
        Research
      </a>
      <a
        className="link"
        href="undefined/press"
      >
        Press
      </a>
      <a
        className="link"
        href="undefined/iconography"
      >
        Iconography
      </a>
      <a
        className="link"
        href="undefined/map"
      >
        Map
      </a>
      <a
        className="link active"
        href="/"
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        Narratives
      </a>
    </div>
  </div>
  <div
    className="container"
    style={
      Object {
        "marginTop": 30,
      }
    }
    text={true}
  >
    <div
      as="a"
      className="button"
      content="My narratives / Minhas narrativas"
      href="/projects"
      icon="angle left"
    />
    <h1
      as="h1"
      className="header"
    >
      Edit Narrative
    </h1>
    <form
      className="form"
      loading={false}
    >
      <input
        className="input"
        icon="linkify"
        label="URL"
        readOnly={true}
        value="http://localhost/"
      />
      <div
        className="field"
      >
        <input
          checked={true}
          className="checkbox"
          label="Submit for publication approval"
          onChange={[Function]}
        />
      </div>
      <hr
        className="divider"
        style={
          Object {
            "margin": "40px 0",
          }
        }
      />
      <div
        className="field"
        required={true}
      >
        <label>
          Title
        </label>
        <input
          className="input"
          onChange={[Function]}
          value="Test Map"
        />
      </div>
      <div
        className="field"
      >
        <label>
          Description
        </label>
        <textarea
          className="tinymce"
        />
      </div>
      <div
        className="field"
      >
        <label>
          Image
        </label>
        <div
          className="segment"
        >
          <img
            className="img"
            src="https://rio-narrative-files.s3.amazonaws.com/"
          />
          <div
            circular={true}
            className="button"
            icon="close"
            onClick={[Function]}
            size="mini"
          />
          <div
            className="field"
            disabled={false}
            inline={true}
            style={
              Object {
                "marginTop": 15,
              }
            }
          >
            <label
              className="inlineLabel"
            >
              Caption
            </label>
            <input
              className="input"
              onChange={[Function]}
              placeholder="Image caption"
              value="Test image"
            />
          </div>
          <div
            className="field"
            disabled={false}
            error={false}
            inline={true}
          >
            <label
              className="inlineLabel"
            >
              Link
            </label>
            <input
              className="input"
              onChange={[Function]}
              placeholder="https://..."
              value="Test source"
            />
          </div>
        </div>
      </div>
      <div
        className="field"
      >
        <label>
          Tags
        </label>
        <select
          allowAdditions={true}
          className="dropdown"
          fluid={true}
          multiple={true}
          onAddItem={[Function]}
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "2",
                "text": "Test",
                "value": "2",
              },
            ]
          }
          placeholder="Select tags"
          search={true}
          selection={true}
          value={
            Array [
              "2",
            ]
          }
        />
      </div>
      <div
        className="field"
      >
        <label>
          Category
        </label>
        <select
          className="dropdown"
          fluid={true}
          onChange={[Function]}
          options={
            Array [
              Object {
                "key": "null",
                "text": "",
                "value": null,
              },
              Object {
                "key": "Literature",
                "text": "Literature",
                "value": "Literature",
              },
            ]
          }
          placeholder="Select a category"
          selection={true}
          value="Literature"
        />
      </div>
      <div
        className="button"
        disabled={false}
        floated="right"
        onClick={[Function]}
        primary={true}
        size="big"
        style={
          Object {
            "paddingLeft": 60,
            "paddingRight": 60,
          }
        }
        type="submit"
      >
        Save
      </div>
      <div
        className="button"
        disabled={false}
        floated="right"
        href="/projects"
        size="big"
        style={
          Object {
            "marginRight": 20,
          }
        }
      >
        Cancel
      </div>
      <div
        style={
          Object {
            "clear": "left",
            "margin": 100,
          }
        }
      />
    </form>
    <div
      basic={true}
      className="modal"
      onClose={[Function]}
      onOpen={[Function]}
      open={false}
      size="small"
      trigger={
        <Button
          content="Delete Project"
          disabled={false}
          fluid={true}
          icon="trash"
          labelPosition="left"
          negative={true}
        />
      }
    >
      <h1
        className="header"
        icon={true}
      >
        <div
          className="icon"
          name="trash"
        />
        Delete this narrative?
      </h1>
      <div
        className="content"
      >
        <p>
          Are you sure you want to delete this narrative? This action is permanent and cannot be undone.
        </p>
      </div>
      <div
        className="actions"
      >
        <div
          basic={true}
          className="button"
          color="red"
          inverted={true}
          onClick={[Function]}
        >
          <div
            className="icon"
            name="remove"
          />
          <span>
            No
          </span>
        </div>
        <div
          className="button"
          inverted={true}
          negative={true}
          onClick={[Function]}
        >
          <div
            className="icon"
            name="trash"
          />
          <span>
            Yes
          </span>
        </div>
      </div>
    </div>
    <img
      className="img"
      src="/img/hrc-logo.png"
      style={
        Object {
          "marginTop": 60,
        }
      }
    />
  </div>
</div>
`;
